### PR TITLE
[wasm] Clean up, and consolidate `tests.wasm.targets`, `tests.wasi.targets`, and `tests.browser.targets`

### PR DESCRIPTION
**Pull request by radical**
Consolidates `tests.wasm.targets`, `tests.wasi.targets`, and `tests.browser.targets` and removes duplication.

### CODE REVIEW

File: eng/testing/tests.browser.targets [hunk 1]
```diff
@@ -1,69 +1,47 @@
 <Project TreatAsLocalProperty="ArchiveTests">
+  <PropertyGroup>
+    <_UseWasmSymbolicator Condition="'$(TestTrimming)' != 'true'">true</_UseWasmSymbolicator>
+  </PropertyGroup>
+
   <Import Project="tests.wasm.targets" />
   <!-- We need to set this in order to get extensibility on xunit category traits and other arguments we pass down to xunit via MSBuild properties -->
   <PropertyGroup>
     <IsBrowserWasmProject Condition="'$(IsBrowserWasmProject)' == ''">true</IsBrowserWasmProject>
-    <WasmGenerateAppBundle Condition="'$(WasmGenerateAppBundle)' == ''">true</WasmGenerateAppBundle>
```
Comment (ilonatommy) [inline eng/testing/tests.browser.targets:6]: I focused mainly on removals. So far that's what I understood:
- `WasmGenerateAppBundle` defaults to true anyway, ok
- `ArchiveTests` - this I don't understand, it's used e.g. by `src/libraries/tests.proj`, won't deleting it change the behavior?
- `BundleTestAppTargets` - looks like it's never read, was only set, ok
- `DebuggerSupport` - is there a sense of letting `DebuggerSupport` to be true for Release?

**Comment by pavelsavara** [comment]
@radical 
- could you please enumerate names of tasks and properties you deleted because they are obsolete and don't need replacement ?
- also list of tasks which just moved and to where they moved

It would help us to make navigate the PR better. Many thanks!

**Comment by radical** [comment]
@ilonatommy @pavelsavara Initially we had `tests.wasm.targets` only. Then with my wasi PR, I had made copies of that as `tests.wasi.targets`, and `tests.browser.targets` and made some changes to make sure it was working. But these had lot of duplications. With this PR, I'm simply consolidating the changes, so not really removing anything but just moving them around so `tests.wasm.targets` is the base file. And `tests.wasi.targets`, and `tests.browser.targets` "derive" from them.

So, the common parts stay in `tests.wasm.targets`, and the other two override what's needed. But it's not perfect, and will be changing as we implement more features for the wasi build.

**Review by pavelsavara — APPROVED** [review]

**Comment by radical** [comment]
thank you for the reviews, @pavelsavara and @ilonatommy !

**Pull request merged**